### ISSUE TITLE
Script ran successfully with MinIO 4.0.7 nuget package failed with MinIO 5.0.0

### ISSUE THREAD

**Issue by ebozduman** [minio/minio-dotnet#864]
Script ran successfully with MinIO 4.0.7 nuget package failed with MinIO 5.0.0

Error log messages:
```
2023-09-04 11:50:17,202 [10] ERROR HPP.FileIT.COBO.UploadInterface.ReverseFeedService.MoveNext[0] - [Bucket]  Exception: System.Threading.Tasks.TaskCanceledException: The operation was canceled.
---> System.IO.IOException: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request..
---> System.Net.Sockets.SocketException (995): The I/O operation has been aborted because of either a thread exit or an application request.
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.Net.Security.SslStream.EnsureFullTlsFrameAsync[TIOAdapter](TIOAdapter adapter)
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer)
   at System.Net.Http.HttpConnection.FillAsync(Boolean async)
   at System.Net.Http.HttpConnection.CopyToContentLengthAsync(Stream destination, Boolean async, UInt64 length, Int32 bufferSize, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnection.ContentLengthReadStream.CompleteCopyToAsync(Task copyTask, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.ContentLengthReadStream.CompleteCopyToAsync(Task copyTask, CancellationToken cancellationToken)
   at Minio.MinioClient.<>c__DisplayClass72_0.<<GetObjectFileAsync>b__0>d.MoveNext() in /_/Minio/Helper/OperationsHelper.cs:line 81
--- End of stack trace from previous location ---
   at Minio.MinioClient.<>c__DisplayClass72_0.<<GetObjectFileAsync>b__1>d.MoveNext() in /_/Minio/Helper/OperationsHelper.cs:line 91
--- End of stack trace from previous location ---
   at Minio.MinioClient.ExecuteTaskCoreAsync(IEnumerable`1 errorHandlers, HttpRequestMessageBuilder requestMessageBuilder, Boolean isSts, CancellationToken cancellationToken) in /_/Minio/MinioClient.cs:line 438
   at Minio.MinioClient.GetObjectStreamAsync(GetObjectArgs args, ObjectStat objectStat, Func`3 cb, CancellationToken cancellationToken) in /_/Minio/Helper/OperationsHelper.cs:line 135
   at Minio.MinioClient.GetObjectHelper(GetObjectArgs args, CancellationToken cancellationToken) in /_/Minio/Helper/OperationsHelper.cs:line 49
   at HPP.FileIT.COBO.UploadInterface.ReverseFeedService.downloadObject(MinioClient minio, String bucketName, String objectName, String downloadFile, IServerSideEncryption sse)
```

and the script is:
```
using System;
using Minio;
using Minio.Credentials;
using Minio.DataModel;
using System.Configuration;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Net.Security;
using System.Runtime.InteropServices;
using System.Security.Cryptography.X509Certificates;
using System.Text;
using System.Threading.Tasks;
using System.Web;
using System.Xml.Serialization;

namespace MinIoDemo
{
    class Program
    {
        static void Main(string[] args)
        {
            try
            {
                Run().Wait();
            }
            catch (Exception e)
            {
                Console.WriteLine($"CertificateIdentityExample test exception: {e}");
                Console.ReadKey();
            }
        }

        private static async Task Run()
        {
            var endpoint = ConfigurationManager.AppSettings["EndPoint"]; //"play.min.io";
            var bucketName = ConfigurationManager.AppSettings["BucketName"];
            var downloadObjectName = ConfigurationManager.AppSettings["DownloadObjectName"];
            var uploadObjectName = ConfigurationManager.AppSettings["UploadObjectName"];
            var secure = true;
            var fileName = ConfigurationManager.AppSettings["filepath"];    // for upload and download
            var RemoteAddress = ConfigurationManager.AppSettings["RemoteAddress"];
            var Port = ConfigurationManager.AppSettings["Port"];
            using (var cert = new X509Certificate2(ConfigurationManager.AppSettings["CertificatePath"], "manage"))  // using (var cert = new X509Certificate2(ConfigurationManager.AppSettings["CertificatePath"], "manage"))
            {
                try
                {
                    var provider = new CertificateIdentityProvider()
                        .WithStsEndpoint(endpoint)
                        .WithCertificate(cert).Build();

                    MinioClient minioClient = new MinioClient()
                        .WithEndpoint(RemoteAddress + ":" + Port).WithSSL()
                        .WithCredentialsProvider(provider).Build();

                   downloadObject(minioClient, bucketName, downloadObjectName, fileName).Wait();
                   Console.WriteLine("\nCertificateIdentityProvider test PASSed\n");
                   Console.ReadKey();
                }
                catch (Exception e)
                {
                    Console.WriteLine($"\nCertificateIdentityProvider test exception: {e}\n");
                    Console.ReadKey();
                }
            }
        }

        private static async Task downloadObject(MinioClient minio, string bucketName, string objectName,
        string downloadFile)
        {
            try
            {
                Console.WriteLine("Running example for API: GetObjectAsync");
                var args = new GetObjectArgs()
                    .WithBucket(bucketName)
                    .WithObject(objectName)
                    .WithFile(downloadFile);
                var stat = await minio.GetObjectAsync(args);
                Console.WriteLine($"Downloaded the file {downloadFile} from bucket {bucketName}");
                // Console.WriteLine($"Stat details of object {objectName} in bucket {bucketName}\n" + stat);
                Console.WriteLine();
            }
            catch (Exception e)
            {
                Console.WriteLine($"[Bucket]  Exception: {e}");
            }
        }
    }
}
```

**Comment by martijn00** [comment]
@ebozduman `Run().Wait();` is a very bad and dangerous call.

Try using `static Task Main(string[] args)` and await the method. Also add ConfigureAwait(false) to it.

What is the result after that change?

**Comment by ebozduman** [comment]
@martijn00 ,

The script came from the customer. I tried to keep their script the same as much as possible but those parts changed in my reproduction script and I was able to reproduce it.
The fix is in #866.
Please check it out and review.
